pennylane/capture/meta_type.py

```diff
@@ -124,6 +165,53 @@ def _(*_, **__):
     return primitive
 
 
+def create_measurement_obs_primitive(
+    measurement_type: type, name: str
+) -> Optional["jax.core.Primitive"]:
+    """Create a primitive corresponding to the input type where the abstract inputs are an operator."""
+    if not has_jax:
+        return None
+
+    primitive = jax.core.Primitive(name)
+
+    @primitive.def_impl
+    def _(obs, **kwargs):
+        return measurement_type(obs=obs, **kwargs)
```

Comment:
This creates a recursive loop to bind this primitive. I think we want to instead use `type.__call__` here, similar to the operator primitive.

```suggestion
        return type.__call__(measurement_type, obs=obs, **kwargs)
```